src/controls/QskListViewSkinlet.cpp

```diff
@@ -110,6 +110,51 @@ QSGNode* QskListViewSkinlet::updateContentsNode(
     return listViewNode;
 }
 
+void QskListViewSkinlet::updateBackgroundNode( const QskListView* listView, QSGNode* rowNode, const QRectF& boxRect, const int row ) const
+{
+    const auto selectedRow = listView->selectedRow();
+    const auto subControl = listView->rowSubControl( row );
+
+    // current row is a selected row
+    if ( row == selectedRow )
+    {
+        auto* boxNode = rowNode->firstChild();
+        auto prepend = boxNode == nullptr;
+
+        QskSkinStateChanger stateChanger( listView );
+        stateChanger.setStates( listView->skinStates() | QskListView::Selected );
+        boxNode = updateBoxNode( listView, boxNode, boxRect, subControl );
```

Comment:
this call to `updateBoxNode`can fail such that it returns `nullptr` e.g. when the box color/gradient hint is transparent or the rectangle is empty

@uwerat maybe naming these functions to `updateOrCreateBoxNode` would help the communicate their behavior. Especially since we give the function a node pointer ( that can be nullptr ) and get a pointer back. Is a passed in non-null pointer always the returned pointer?